src/entities/elastic-storage/elastic-storage-processor.js

```diff
@@ -450,15 +460,125 @@ class ElasticStorageProcessor<TxType: ByronTxType | ShelleyTxType> implements St
     return utxoAddresses
   }
 
+  async storeGenesisBlockData(block: Array<Block>): Promise<void> {
+    const chunk = ++this.lastChunk
+    const utxosForInputsAndOutputs = getBlockUtxos(block)
+    const blockOutputsToStore = utxosForInputsAndOutputs
+    const blockTxsCount = block.getTxs().length
+    const mappedBlocks: Array<BlockData> = getBlocksForSlotIdx(
+      [block],
+      [],
+      {},
+      {},
+      {},
+    )
+
+
+    const tip: BlockInfoType = await this.getBestBlockNum()
+    const paddedBlocks: Array<BlockData> = padEmptySlots(mappedBlocks,
+      tip.epoch, tip.slot, this.networkStartTime, this.slotsPerEpoch)
+
+    const blocksData = paddedBlocks.map((b: BlockData) => b.toPlainObject(true))
+
+    const isGenesisBlock = true
+    this.logger.debug(`storeBlocksData.constructing bulk for ${blocksData.length} slots`)
+    const blocksBody = formatBulkUploadBody(blocksData, {
+      index: this.indexFor(INDEX_SLOT),
+      getId: (o) => o.hash,
+      getData: o => ({
+        ...o,
+        ...(isGenesisBlock ? {
+          tx: undefined,
+        } : {}),
+        _chunk: chunk,
+      }),
+    })
+
+    const blockTxioToStore = blockOutputsToStore
+    this.logger.debug(`storeBlocksData.constructing bulk for ${blockTxioToStore.length} txio`)
+    const txiosBody = formatBulkUploadBody(blockTxioToStore, {
+      index: this.indexFor(INDEX_TXIO),
+      getId: (o) => o.id,
+      getData: (o) => ({
+        ...o,
+        _chunk: chunk,
+      }),
+    })
+
+    const bulkData = [...blocksBody, ...txiosBody]
+    this.logger.debug(`storeBlocksData.total bulk is ${bulkData.length} documents`)
+    const bulkChunks = _.chunk(bulkData, 1000)
+    for (let i = 0; i < bulkChunks.length; i += 1) {
+      this.logger.debug(`storeBlocksData.bulkUpload chunk ${i + 1} out of ${bulkChunks.length}`)
+      try {
+        await this.bulkUpload(bulkChunks[i])
+        await sleep(100)
+      } catch (e) {
+        this.logger.error(`Failed to bulk-upload blocks data (chunk ${i + 1} out of ${bulkChunks.length})`, e)
+        throw new Error(`Failed to bulk-upload blocks data (chunk ${i + 1} out of ${bulkChunks.length}) : ${e}`)
+      }
+    }
+
+    for (const b of paddedBlocks) {
+      // const txsData = []
+      let idx = 0
+      for (const txData of b.getTxsData()) {
+        //txsData.push(txData)
+        // collect 1k items and upload them
+        // if (idx % 1000 === 0) {
+        const txsBody = formatBulkUploadBody([txData], {
+            index: this.indexFor(INDEX_TX),
+            getId: (o) => o.hash,
+            getData: (o) => o,
+          })
+          await sleep(100)
+          await this.bulkUpload(txsBody)
+          // txsData.splice(0, txsData.length)
+        // }
+        ++idx
+        this.logger.debug('Idx:', idx)
+      }
+    }
+
+    /*
+    const txsDataGen = paddedBlocks.flatMap(b => b.getTxsData())
```

Comment:
do we need this commented code? The code above also has some commented code in it.